app/queries/get_deferred_application_choices_for_current_cycle.rb

```diff
@@ -0,0 +1,8 @@
+class GetDeferredApplicationChoicesForCurrentCycle
+  def self.call
+    ApplicationChoice
+      .joins(:course)
+      .where(status: :offer_deferred)
+      .where('courses.recruitment_cycle_year': RecruitmentCycle.previous_year)
```

Comment:
I'd recommend adding a service spec for this, considering the button triggers multiple emails. Also, we'll soon be changing the way we target these deferred offers, finding them based on `offer_deferred_at` rather than the course recruitment_cycle_year.